### ISSUE TITLE
[nrfconnect] Remove temporary timeval problem fix after merging solution from Zephyr.

### ISSUE THREAD

**Issue by kkasperczyk-no** [project-chip/connectedhomeip#3822]
#### Problem
Temporary solution was introduced in order to fix timeval type size problem resulting in improper operation of the select method. It's an internal Zephyr implementation problem and it will be fixed there.

 #### Proposed Solution
Remove changes introduced in https://github.com/project-chip/connectedhomeip/pull/3800 after merging fix from Zephyr project.

**Comment by woody-apple** [comment]
@kkasperczyk-no has this been done yet? any ETA?

**Comment by kkasperczyk-no** [comment]
@woody-apple It can't be removed yet. Fix was merged to Zephyr master quite long ago, but in CHIP we got Zephyr revision associated with the nRF Connect SDK version and currently recommended 1.4.0 version does not have it. New version 1.5 of nRF Connect SDK should be released in next few weeks and after that we will update it in CHIP. I remember about that issue, but thank you for reminding and tracking issues status!
